pkg/build/install_hooks.go

```diff
@@ -0,0 +1,56 @@
+package build
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/airplanedev/lib/pkg/utils/fsx"
+	"github.com/pkg/errors"
+)
+
+const (
+	preInstallScriptName  = "airplane_preinstall.sh"
+	postInstallScriptName = "airplane_postinstall.sh"
+)
+
+type InstallHooks struct {
+	// Paths are relative to the task's root
+	PreInstallFilePath  string
+	PostInstallFilePath string
+}
+
+// GetInstallHooks look for install scripts in every directory from the entrypoint's
+// directory up to the root directory.
+func GetInstallHooks(entrypoint string, root string) (InstallHooks, error) {
+	find := func(filename string) (string, error) {
+		if start, found := fsx.FindUntil(
+			// Entrypoint path is relative to the root so we need to combine them.
+			filepath.Join(root, filepath.Dir(entrypoint)),
+			filepath.Dir(root),
+			filename,
+		); found {
```

Comment:
It's idiomatic to keep the happy path unindented, and put error handling / sad path handling in these if clauses. 

```suggestion
		); !ok {
```